lib/ansible/modules/cloud/amazon/ec2_eni.py

```diff
@@ -378,161 +492,237 @@ def modify_eni(connection, vpc_id, module, eni):
+                )
                 changed = True
-        if delete_on_termination is not None and eni.attachment is not None:
-            if eni.attachment.delete_on_termination is not delete_on_termination:
-                connection.modify_network_interface_attribute(eni.id, "deleteOnTermination", delete_on_termination, eni.attachment.id)
+        if delete_on_termination is not None and "Attachment" in eni:
+            if eni["Attachment"]["DeleteOnTermination"] is not delete_on_termination:
+                connection.modify_network_interface_attribute(
+                    NetworkInterfaceId=eni["NetworkInterfaceId"],
+                    Attachment={'AttachmentId': eni["Attachment"]["AttachmentId"],
+                                'DeleteOnTermination': delete_on_termination}
+                )
                 changed = True
 
-        current_secondary_addresses = [i.private_ip_address for i in eni.private_ip_addresses if not i.primary]
+        current_secondary_addresses = []
+        if "PrivateIpAddresses" in eni:
+            current_secondary_addresses = [i["PrivateIpAddress"] for i in eni["PrivateIpAddresses"] if not i["Primary"]]
+
         if secondary_private_ip_addresses is not None:
             secondary_addresses_to_remove = list(set(current_secondary_addresses) - set(secondary_private_ip_addresses))
             if secondary_addresses_to_remove and purge_secondary_private_ip_addresses:
-                connection.unassign_private_ip_addresses(network_interface_id=eni.id,
-                                                         private_ip_addresses=list(set(current_secondary_addresses) -
-                                                                                   set(secondary_private_ip_addresses)),
-                                                         dry_run=False)
+                connection.unassign_private_ip_addresses(
+                    NetworkInterfaceId=eni["NetworkInterfaceId"],
+                    PrivateIpAddresses=list(set(current_secondary_addresses) - set(secondary_private_ip_addresses)),
+                )
                 changed = True
 
             secondary_addresses_to_add = list(set(secondary_private_ip_addresses) - set(current_secondary_addresses))
             if secondary_addresses_to_add:
-                connection.assign_private_ip_addresses(network_interface_id=eni.id,
-                                                       private_ip_addresses=secondary_addresses_to_add,
-                                                       secondary_private_ip_address_count=None,
-                                                       allow_reassignment=allow_reassignment, dry_run=False)
+                connection.assign_private_ip_addresses(
+                    NetworkInterfaceId=eni["NetworkInterfaceId"],
+                    PrivateIpAddresses=secondary_addresses_to_add,
+                    AllowReassignment=allow_reassignment
```

Comment:
The Amazon docs state that reassignment is asynchronous and you need to wait for it.  Do we want to handle that here?  https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_AssignPrivateIpAddresses.html

Reply:
Yes, I'll add a wait.